### ISSUE TITLE
--skip-rule doesn't work with parser warnings

### ISSUE THREAD

**Issue by Amachua** [stoplightio/spectral#1550]
**Describe the bug**

Spectral version used: "5.8.0"

In my perspecitive, there is possibly two distinct issues here:
- the tool forward issues `parser` when a JSON file includes tabulation (which seems to be allowed with the file I've provided;
- the `skip-rule` parameter doesn't apply to the `parser` warning.

**To Reproduce**

Preferred solution: 

Given the file [resources.openapi.txt](https://github.com/stoplightio/spectral/files/6140230/resources.openapi.txt)

When I run the command line

> yarn spectral lint .\resources.openapi.txt

Then only the following issue is prompted:

```text
 18:13  warning  operation-operationId  Operation should have an `operationId`.  paths./dummy/v1/resources.get

X 1 problem (0 errors, 1 warning, 0 infos, 0 hints)

Done in 2.50s.
```

Other solution: 

Given the file [resources.openapi.txt](https://github.com/stoplightio/spectral/files/6140230/resources.openapi.txt)

When I run the command line

> yarn spectral lint .\publish\resources.openapi.txt --skip-rule parser

Then only the following issue is prompted:

```text
 18:13  warning  operation-operationId  Operation should have an `operationId`.  paths./dummy/v1/resources.get

X 1 problem (0 errors, 1 warning, 0 infos, 0 hints)

Done in 2.50s.
```

FWIW, the current output I receive is:

```text
  2:1   warning  parser                 Using tabs can lead to unpredictable results
  3:1   warning  parser                 Using tabs can lead to unpredictable results
  4:1   warning  parser                 Using tabs can lead to unpredictable results
  4:2   warning  parser                 Using tabs can lead to unpredictable results
  5:1   warning  parser                 Using tabs can lead to unpredictable results
  5:2   warning  parser                 Using tabs can lead to unpredictable results
  5:3   warning  parser                 Using tabs can lead to unpredictable results
  6:1   warning  parser                 Using tabs can lead to unpredictable results
  6:2   warning  parser                 Using tabs can lead to unpredictable results
  7:1   warning  parser                 Using tabs can lead to unpredictable results
  8:1   warning  parser                 Using tabs can lead to unpredictable results
  9:1   warning  parser                 Using tabs can lead to unpredictable results
  9:2   warning  parser                 Using tabs can lead to unpredictable results
  10:1  warning  parser                 Using tabs can lead to unpredictable results
  10:2  warning  parser                 Using tabs can lead to unpredictable results
  11:1  warning  parser                 Using tabs can lead to unpredictable results
  11:2  warning  parser                 Using tabs can lead to unpredictable results
  12:1  warning  parser                 Using tabs can lead to unpredictable results
  12:2  warning  parser                 Using tabs can lead to unpredictable results
  12:3  warning  parser                 Using tabs can lead to unpredictable results
  13:1  warning  parser                 Using tabs can lead to unpredictable results
  13:2  warning  parser                 Using tabs can lead to unpredictable results
  14:1  warning  parser                 Using tabs can lead to unpredictable results
  14:2  warning  parser                 Using tabs can lead to unpredictable results
  15:1  warning  parser                 Using tabs can lead to unpredictable results
  16:1  warning  parser                 Using tabs can lead to unpredictable results
  17:1  warning  parser                 Using tabs can lead to unpredictable results
  17:2  warning  parser                 Using tabs can lead to unpredictable results
  18:1  warning  parser                 Using tabs can lead to unpredictable results
  18:2  warning  parser                 Using tabs can lead to unpredictable results
  18:3  warning  parser                 Using tabs can lead to unpredictable results
 18:10  warning  operation-operationId  Operation should have an `operationId`.       paths./dummy/v1/resources.get
  19:1  warning  parser                 Using tabs can lead to unpredictable results
  19:2  warning  parser                 Using tabs can lead to unpredictable results
  19:3  warning  parser                 Using tabs can lead to unpredictable results
  19:4  warning  parser                 Using tabs can lead to unpredictable results
  20:1  warning  parser                 Using tabs can lead to unpredictable results
  20:2  warning  parser                 Using tabs can lead to unpredictable results
  20:3  warning  parser                 Using tabs can lead to unpredictable results
  20:4  warning  parser                 Using tabs can lead to unpredictable results
  21:1  warning  parser                 Using tabs can lead to unpredictable results
  21:2  warning  parser                 Using tabs can lead to unpredictable results
  21:3  warning  parser                 Using tabs can lead to unpredictable results
  21:4  warning  parser                 Using tabs can lead to unpredictable results
  22:1  warning  parser                 Using tabs can lead to unpredictable results
  22:2  warning  parser                 Using tabs can lead to unpredictable results
  22:3  warning  parser                 Using tabs can lead to unpredictable results
  22:4  warning  parser                 Using tabs can lead to unpredictable results
  22:5  warning  parser                 Using tabs can lead to unpredictable results
  23:1  warning  parser                 Using tabs can lead to unpredictable results
  23:2  warning  parser                 Using tabs can lead to unpredictable results
  23:3  warning  parser                 Using tabs can lead to unpredictable results
  23:4  warning  parser                 Using tabs can lead to unpredictable results
  24:1  warning  parser                 Using tabs can lead to unpredictable results
  24:2  warning  parser                 Using tabs can lead to unpredictable results
  24:3  warning  parser                 Using tabs can lead to unpredictable results
  24:4  warning  parser                 Using tabs can lead to unpredictable results
  25:1  warning  parser                 Using tabs can lead to unpredictable results
  25:2  warning  parser                 Using tabs can lead to unpredictable results
  25:3  warning  parser                 Using tabs can lead to unpredictable results
  25:4  warning  parser                 Using tabs can lead to unpredictable results
  25:5  warning  parser                 Using tabs can lead to unpredictable results
  26:1  warning  parser                 Using tabs can lead to unpredictable results
  26:2  warning  parser                 Using tabs can lead to unpredictable results
  26:3  warning  parser                 Using tabs can lead to unpredictable results
  26:4  warning  parser                 Using tabs can lead to unpredictable results
  26:5  warning  parser                 Using tabs can lead to unpredictable results
  26:6  warning  parser                 Using tabs can lead to unpredictable results
  27:1  warning  parser                 Using tabs can lead to unpredictable results
  27:2  warning  parser                 Using tabs can lead to unpredictable results
  27:3  warning  parser                 Using tabs can lead to unpredictable results
  27:4  warning  parser                 Using tabs can lead to unpredictable results
  27:5  warning  parser                 Using tabs can lead to unpredictable results
  28:1  warning  parser                 Using tabs can lead to unpredictable results
  28:2  warning  parser                 Using tabs can lead to unpredictable results
  28:3  warning  parser                 Using tabs can lead to unpredictable results
  28:4  warning  parser                 Using tabs can lead to unpredictable results
  28:5  warning  parser                 Using tabs can lead to unpredictable results
  29:1  warning  parser                 Using tabs can lead to unpredictable results
  29:2  warning  parser                 Using tabs can lead to unpredictable results
  29:3  warning  parser                 Using tabs can lead to unpredictable results
  29:4  warning  parser                 Using tabs can lead to unpredictable results
  29:5  warning  parser                 Using tabs can lead to unpredictable results
  29:6  warning  parser                 Using tabs can lead to unpredictable results
  30:1  warning  parser                 Using tabs can lead to unpredictable results
  30:2  warning  parser                 Using tabs can lead to unpredictable results
  30:3  warning  parser                 Using tabs can lead to unpredictable results
  30:4  warning  parser                 Using tabs can lead to unpredictable results
  30:5  warning  parser                 Using tabs can lead to unpredictable results
  31:1  warning  parser                 Using tabs can lead to unpredictable results
  31:2  warning  parser                 Using tabs can lead to unpredictable results
  31:3  warning  parser                 Using tabs can lead to unpredictable results
  31:4  warning  parser                 Using tabs can lead to unpredictable results
  32:1  warning  parser                 Using tabs can lead to unpredictable results
  32:2  warning  parser                 Using tabs can lead to unpredictable results
  32:3  warning  parser                 Using tabs can lead to unpredictable results
  33:1  warning  parser                 Using tabs can lead to unpredictable results
  33:2  warning  parser                 Using tabs can lead to unpredictable results
  34:1  warning  parser                 Using tabs can lead to unpredictable results
  35:1  warning  parser                 Using tabs can lead to unpredictable results
  36:1  warning  parser                 Using tabs can lead to unpredictable results
  36:2  warning  parser                 Using tabs can lead to unpredictable results
  37:1  warning  parser                 Using tabs can lead to unpredictable results
  37:2  warning  parser                 Using tabs can lead to unpredictable results
  37:3  warning  parser                 Using tabs can lead to unpredictable results
  38:1  warning  parser                 Using tabs can lead to unpredictable results
  38:2  warning  parser                 Using tabs can lead to unpredictable results
  39:1  warning  parser                 Using tabs can lead to unpredictable results

X 108 problems (0 errors, 108 warnings, 0 infos, 0 hints)

Done in 3.79s.
```

**Comment by P0lip** [comment]
Yeah `--skip-rule` does not let you do that.
Is there any problem with converting tabs to spaces?

**Comment by Amachua** [comment]
Since the file we're trying to lint are generated from an external tool, each time we re-export we have to convert them to ensure we don't have any `parser` issue.

This work is performed several times per day for something that is conform from the JSON point of view (I suppose that, at some point, other people will face the same issue). Is there a way I can help to solve this issue? (I saw that WoH mentioned this issue on another Git repository, would you like me to do some lobbying there?)

**Comment by P0lip** [comment]
> Since the file we're trying to lint are generated from an external tool, each time we re-export we have to convert them to ensure we don't have any parser issue.

Did you consider incorporating an additional tool for that conversation before you pass that file forward to Spectral?
I think that prettifiers such as Prettier would convert these tabs to spaces if a proper configuration is provided.

> Is there a way I can help to solve this issue?

Theoretically this isn't an issue. Although Spectral is relatively flexible, its certain behavior has limited level of configuration. 
This is a conscious decision, and for the time being I'm not sure if we'll change it (unless it's a really common pattern facing a notable amount of users)
I'm aware this may come as disappointing answer, but this is how things look like for now.
On a positive note - we're planning some changes in v6 around parsing errors and rulesets. A "side effect" of these upcoming changes may be the ability to override a severity of a particular parsing error/warning.

**Comment by Amachua** [comment]
> Did you consider incorporating an additional tool for that conversation before you pass that file forward to Spectral?
> I think that prettifiers such as Prettier would convert these tabs to spaces if a proper configuration is provided.

I've consider some work around to make our pipeline work properly. Nonetheless, I still consider that we have an issue with the Spectral tool but not necessarily with the `--skip-rule ` component since... The more I think on it and the more I think the issue comes from the parser the tool used.

Indeed, while reviewing the [JSON Standard](https://www.json.org/json-en.html), the horizontal tab is part of the allowed whitespace (scroll down in the page) --> It shouldn't throw any issue when finding this character! Also, as a linter of JSON/YAML file, I think that Spectral shouldn't throw any issue when it processed the file I've shared (since they're compliant with the standard).

>  On a positive note - we're planning some changes in v6 around parsing errors and rulesets. A "side effect" of these upcoming changes may be the ability to override a severity of a particular parsing error/warning.

Looking forward onto this! :) But... Still think the issue I've raised is, at the end, not only related to the `--skip-rule ` component and something must be performed on the parser side so it doesn't generates issues with well formatted files.
